scoring/cscore.metta

```diff
@@ -75,8 +103,26 @@
 (: cScore< (-> Cscore Cscore Bool)) 
 (= (cScore< $cs1 $cs2)
     (let ($pl $pr $cl $cr) ((getPenScore $cs1) (getPenScore $cs2) (getComp $cs1) (getComp $cs2))
-        (if (isnan $pl)
-            (not (isnan $pr))
+        (if (type-cast (isnan $pl) Bool &self)
+            (not (type-cast (isnan $pr) Bool &self))
+            (or (< $pl $pr) (and (== $pl $pr) (> $cl $cr))))))
+
+;; cScoreExpr< -- composite score less than operator for expressions
+;; works directly with expressions using index-atom to extract scores
+;; index 4 = penalized score, index 1 = complexity score
+; (: cScoreExpr< (-> Expression Expression Bool))
+(= (cScoreExpr< $cs1 $cs2)
```

Comment:
You are sure that the `cScoreExpr` will have certain type of pattern, so best if you can leverage that and simplfiy the code.